x/compliance/keeper/msg_server_update_compliance_info.go

```diff
@@ -0,0 +1,97 @@
+package keeper
+
+import (
+	"context"
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/x/compliance/types"
+	dclauthtypes "github.com/zigbee-alliance/distributed-compliance-ledger/x/dclauth/types"
+)
+
+func (k msgServer) UpdateComplianceInfo(goCtx context.Context, msg *types.MsgUpdateComplianceInfo) (*types.MsgUpdateComplianceInfoResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check if signer has enough rights to update model
+	signerAddr, err := sdk.AccAddressFromBech32(msg.Creator)
+
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid Address: (%s)", err)
+	}
+
+	if !k.dclauthKeeper.HasRole(ctx, signerAddr, dclauthtypes.CertificationCenter) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized, "%s transaction should be "+
+			"signed by an account with the %s role", "MsgUpdateComplianceInfo", dclauthtypes.CertificationCenter)
+	}
+
+	complianceInfo, isFound := k.GetComplianceInfo(ctx, msg.Vid, msg.Pid, msg.SoftwareVersion, msg.CertificationType)
+
+	if !isFound {
+		return nil, types.NewErrComplianceInfoDoesNotExist(msg.Vid, msg.Pid, msg.SoftwareVersion, msg.CertificationType)
+	}
+
+	if msg.CDCertificateId != "" {
+		complianceInfo.CDCertificateId = msg.CDCertificateId
+	}
+
+	if msg.CDVersionNumber != math.MaxUint32 {
```

Comment:
If we decide that `CDVersionNumber` is mutable, I think we can make it a string in the message, so that we can check against empty string for non-set values.